lib/work/proof.ts

```diff
@@ -27,6 +27,52 @@ export class Solution {
     public ExtraNonce2: Bytes,
     public Nonce: UInt32Little,
     public GeneralPurposeBits?: Int32Little) {}
+
+  toJSON() {
+    let json = {
+      share: {
+        timestamp: this.Time.hex,
+        nonce: this.Nonce.hex,
+        extra_nonce_2: this.Nonce.hex
+      },
+      extra_nonce_1: this.ExtraNonce1.hex
+    }
+
+    if (this.GeneralPurposeBits) json.share["bits"] = this.GeneralPurposeBits.hex
+
+    return json
+  }
+
+  static fromJSON(x): Solution | undefined {
```

Comment:
It would be nice if the variable `x` here used an interface to describe the JSON structure required